src/Raven.Server/Config/Categories/SecurityConfiguration.cs

```diff
@@ -142,6 +142,24 @@ public class SecurityConfiguration : ConfigurationCategory
         [ConfigurationEntry("Security.WellKnownIssuerHashes.Admin", ConfigurationEntryScope.ServerWideOnly)]
         public string[] WellKnownIssuerHashes { get; set; }
 
+        [Description("EXPERT: A command or executable to validate a server authentication request. " +
+                     "RavenDB will execute: command [user-arg-1] ... [user-arg-n] <sender-url> <base64-certificate> <errors>. " +
+                     "The executable will return a case-insensitive boolean string through the standard output (e.g. true, false) indicating whether to approve the connection.")]
+        [DefaultValue(null)]
+        [ConfigurationEntry("Security.CertificateValidation.Exec", ConfigurationEntryScope.ServerWideOnly)]
+        public string CertificateValidationExec { get; set; }
+
+        [Description("EXPERT: The optional user arguments for the 'Security.CertificateValidation.Exec' command or executable. The arguments must be escaped for the command line.")]
+        [DefaultValue(null)]
+        [ConfigurationEntry("Security.CertificateValidation.Exec.Arguments", ConfigurationEntryScope.ServerWideOnly)]
+        public string CertificateValidationExecArguments { get; set; }
+
+        [Description("The number of seconds to wait for the Certificate Validation executable to exit. Default: 30 seconds")]
```

Comment:
30 seconds is a _lot_ here, no? We have a pending client connection.
Would it make sense to use 5 sec and diverge from our normal?